starfish/image/_filter/util.py

```diff
@@ -133,17 +133,19 @@ def preserve_float_range(
 
     """
     array = array.copy()
+
     if isinstance(array, xr.DataArray):
         data = array.values
     else:
         data = array
+
     if np.any(data < 0):
-        data[array < 0] = 0
-    if np.any(array > 1):
+        data[data < 0] = 0
```

Comment:
🐛 !

If data is an xarray, we need to convert it to a numpy array to run these comparisons because indexing works differently between the two objects: http://xarray.pydata.org/en/stable/indexing.html#indexing-rules

This function was just never properly tested with an xarray input. np.ndarray[xr.DataArray] does not work.